.github/workflows/MainDistributionPipeline.yml

```diff
@@ -0,0 +1,20 @@
+#
+# This workflow calls the main distribution pipeline from DuckDB to build, test and (optionally) release the extension
+#
+name: Main Extension Distribution Pipeline
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref || '' }}-${{ github.base_ref || '' }}-${{ github.ref != 'refs/heads/main' || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  duckdb-main-pipeline:
+    name: DuckDB Main Pipeline
+    uses: samansmink/extension-workflows/.github/workflows/_extension_distribution.yml@main
+    with:
+      extension_name: quack
+    secrets: inherit
```

Comment:
I think, given we are calling an external controlled workflow, that it's better to be explicit on what secrets are needed

Reply:
And I would think it's better to have some to explicitly opt-in that can default to "all", so that if eventually we discontinue something you get notified.

Reply:
> And I would think it's better to have some to explicitly opt-in that can default to "all", so that if eventually we discontinue something you get notified.

I disagree here, if you want to distribute an extension, the default setting should be to distribute for all platforms. Excluding builds is only for edge cases where you can't get a platform to work with your extension. I think we should nudge people into distributing for all platforms by default.

Having each extension opt-in a list of platforms is just confusing I think. Consider when we add a platform (linux_arm64_gcc4 for example) to the mainpipeline.yml: when a user then updates the workflow version, he will also need to manually update his list to match the new workflow.

Reply:
Other point of view, should this be there at all?
Or should this be handled CMake side somehow?
But then I don't know exactly how, so maybe it's just too complex.

Reply:
I mean, it's strange to say "don't build for windows_x_y", then we update duckdb/extension-template to add "windows_z", and your workflow start failing.

Reply:
> Or should this be handled CMake side somehow?

How would you configure from Cmake which Github Action workers are launched? this config is fed into the workflow matrices.

>  I mean, it's also strange to say "don't build for windows_x_y", then we update duckdb/extension-template to add "windows_z", and your workflow start failing.

well no: the extension-workflows version is pinned. If you update that, and the newly added workflow is failing, you will just notice a new workflow you've never seen before that is now failing, right?

Reply:
discussed over chat, decided on exclude